tutorials/02-Basic Operations.ipynb

```diff
@@ -0,0 +1,1822 @@
+   ],
+   "source": [
+    "math.sqrt(2) ** 2 - 2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cd7f8ae7-5002-4f60-a183-87e730010293",
+   "metadata": {},
+   "source": [
+    "The math library or numpy library is insufficient to use in actual mathematics, than numeric analysis.\n",
+    "\n",
+    "So if you actually want to do math, SymPy is the only option."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0dec3b94-1855-4d3a-8bec-5704624cf8aa",
+   "metadata": {},
+   "source": [
+    "Every SymPy functions, can be used to construct arbitrary mathematical constant that can be generated by mathematical formula."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "3a4c46f2-39be-49c6-90b8-b71b678dcf29",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sin(1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0863998b-12f8-46a1-bd93-6d3a70b5978b",
+   "metadata": {},
+   "source": [
+    "For example, Apery's constant (https://en.wikipedia.org/wiki/Ap%C3%A9ry%27s_constant) is not available in standard math library. \n",
+    "\n",
+    "But if you need it, you can build it your own by using sympy functions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "5c529331-e310-4073-9fa7-b037936efc93",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "zeta(3).evalf()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "01e6e0a7-dc64-46d4-b6b2-8e35ab130ad8",
+   "metadata": {},
+   "source": [
+    "# Parsing"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "20d98bb3-8ae7-4a86-b60d-7d5c3a3a05d7",
+   "metadata": {},
+   "source": [
+    "We have learned how to build the mathematical formula by programming with symbols.\n",
+    "\n",
+    "However, you can alternatively build SymPy expressions by parsing from strings."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "77b53512-205d-4977-bbaa-a4865379107a",
+   "metadata": {},
+   "source": [
+    "The pros of using parser is that:\n",
+    "\n",
+    "- It is easy to store expression as file and read to SymPy (serializing/deserializing).\n",
+    "- It can use more 'mathematical' grammar for expressions than Python:\n",
+    "  - You always need to declare symbols first, and undefined program variables don't become symbols\n",
+    "  - `^` isn't power"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1ee047e3-56d6-400a-a463-37de124c92cd",
+   "metadata": {},
+   "source": [
+    "## SymPy parser"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7796ec0-9188-444c-abf9-03db6a6bc769",
+   "metadata": {},
+   "source": [
+    "`parse_expr` can be used to parse text into SymPy expressions."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e8c3c36b-3d2d-4d3f-9803-4c8e202873cc",
+   "metadata": {},
+   "source": [
+    "`parse_expr` can do the following:\n",
+    "\n",
+    "- Converts `^` into python `**` operator.\n",
+    "- Use factorial with `!`\n",
```

Comment:
I don't think we should mention this. I'd like to remove this feature by default. We don't really need to mention the `^` thing either.